src/App.js

```diff
@@ -2,56 +2,47 @@ import React, { Component } from 'react';
 import './App.css';
 import Posters from './Posters'
 import Movie from './Movie'
-import {fetchMovies, fetchSingleMovie} from './apiCalls';
+import { fetchMovies } from './apiCalls';
 import { Route } from 'react-router-dom';
 
 class App extends Component {
   constructor() {
     super();
     this.state = {
       movies: [],
-      singleMovie: {},
       error: ''
     }
   }
   
   componentDidMount() {
     fetchMovies()
     .then(data => this.setState({movies: data.movies}))
-    .catch(error => this.setState({error: error}))
-  }
-  
-  clickPoster = (id) => {   
-    fetchSingleMovie(id)
-    .then(data => this.setState({singleMovie: data.movie}))
-    .catch(error => this.setState({error: error}))
-  }
-
-  clickBackBtn = () => {
-    this.setState({singleMovie: {}})
+    .catch(error => this.setState({error: error.message}))
   }
 
   render() {
-    const {movies, singleMovie, error} = this.state;
+    const {movies, error} = this.state;
     
     return (
       <main className="App">
         <h1 className="App-header">Rancid Tomatillos</h1>
 
-        {error && <h2>{error}</h2>}
+        {error && <h2>The server doesn't seem to be working right now</h2>} 
         
-        {!movies.length && <p>Hang Tight!</p>}
+        {!movies.length && !error && <p>Hang Tight!</p>}
 
-        <Route exact path='/' render={() => <Posters posters={movies} clickPoster={this.clickPoster} /> } />
-        <Route exact path={`/${singleMovie.id}`} render={() => <Movie movie={singleMovie} clickBackBtn={this.clickBackBtn} /> } />
+        <Route exact path='/' render={() => <Posters posters={movies} /> } />
+        
+        <Route 
+          exact path='/:id' 
+          render={({ match }) => {
```

Comment:
The match will be really great to learn & implement more in the future. I'm so glad that we got this working.